src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -173,7 +173,10 @@ private void AnalyzeExpressionStatement(OperationAnalysisContext context)
                     //  4. Bail out if there is language specific syntax to indicate an explicit discard.
                     //     For example, VB call statement is used to explicitly ignore the value returned by
                     //     an invocation by prefixing the invocation with keyword "Call".
-                    if (_symbolStartAnalyzer._compilationAnalyzer.IsCallStatement(expressionStatement))
+                    //     Similarly, we do not want to flag an expression of a C# expression body.
+                    if (_symbolStartAnalyzer._compilationAnalyzer.IsCallStatement(expressionStatement) ||
+                        (expressionStatement.Parent is IBlockOperation blockParent &&
+                         _symbolStartAnalyzer._compilationAnalyzer.IsExpressionBody(blockParent)))
```

Comment:
💡 Having both `||` and `&&` in the expression is hard to read, especially when they are placed at the end of the line. Any of the following would be better:

1. Make sure the parenthesized `&&` expression fits on one line
2. Move the `&&` expression evaluation to a local function or helper method
3. Place the operators at the beginning of the line instead of the end of the line (not as good as the previous two options, but it at least makes the reader aware of the situation)

Reply:
Thanks, hope https://github.com/dotnet/roslyn/pull/33042/commits/7db8845eca416b9ae32ecb7223d7a2667e250693 improves the readability.